src/bm_sim/P4Objects.cpp

```diff
@@ -339,6 +339,12 @@ P4Objects::process_single_param(ActionFn* action_fn,
       process_single_param(action_fn, cfg_parameter_value, primitive_name);
     }
     action_fn->parameter_end_vector();
+  } else if (type == "field_list") {
+    action_fn->start_field_list();
+    for (const auto &cfg_parameter_value : cfg_parameter["value"]) {
+      process_single_param(action_fn, cfg_parameter_value, primitive_name);
```

Comment:
it feels like we should be able to add a check here to make sure that `cfg_parameter["value"]` is a field reference?